native/mod_proxy_cluster/mod_proxy_cluster.c

```diff
@@ -895,14 +899,26 @@ static proxy_worker *get_worker_from_id_stat(proxy_server_conf *conf, int id, pr
             proxy_worker **worker = (proxy_worker **) ptrw;
             proxy_cluster_helper *helper = (proxy_cluster_helper *) (*worker)->context;
             if ((*worker)->s == stat && helper->index == id) {
+
+                /* Check that the worker is really the one we need */
+                char sport[7];
+                apr_snprintf(sport, sizeof(sport), "%d", (*worker)->s->port);
+                if (strcmp((*worker)->s->scheme, node->mess.Type) ||
+                    strcmp((*worker)->s->hostname, node->mess.Host) ||
+                    strcmp(sport, node->mess.Port)) {
+                    (*worker)->s->index = 0;
+                    mpm_generation--; /* mark old generation that will recreate the process */
```

Comment:
@jfclere I wanted to pull an address of that value via ap_mpm_query and update the value at that address, not to create a local copy :disappointed: Looking into it again...